tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -0,0 +1,52 @@
+// <copyright file="DatabaseMonitoringPropagator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.DatabaseMonitoring
+{
+    internal static class DatabaseMonitoringPropagator
+    {
+        private const string SqlCommentSpanService = "dddbs";
+        private const string SqlCommentRootService = "ddps";
+        private const string SqlCommentVersion = "ddpv";
+        private const string SqlCommentEnv = "dde";
+
+        internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, string configuredServiceName, Span span)
+        {
+            if (propagationStyle == DbmPropagationLevel.Disabled)
+            {
+                return string.Empty;
+            }
+
+            var propagatorSringBuilder = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
+            propagatorSringBuilder.Append($"/*{SqlCommentRootService}='{configuredServiceName}',{SqlCommentSpanService}='{span.ServiceName}'");
+
+            if (span.GetTag(Tags.Version) is { } versionTag)
+            {
+                propagatorSringBuilder.Append($",{SqlCommentVersion}='{versionTag}'");
+            }
+
+            if (span.GetTag(Tags.Env) is { } envTag)
+            {
+                propagatorSringBuilder.Append($",{SqlCommentEnv}='{envTag}'");
+            }
+
+            if (propagationStyle == DbmPropagationLevel.Full)
+            {
+                propagatorSringBuilder.Append($",{W3CTraceContextPropagator.TraceParentHeaderName}='{W3CTraceContextPropagator.CreateTraceParentHeader(span.Context)}'*/");
+            }
+            else
+            {
+                propagatorSringBuilder.Append("*/");
+            }
```

Comment:
Watch our here. Any enum value that is not `Disabled`, `Service`, or `Full` will fall into this branch.
```suggestion
            else if (propagationStyle == DbmPropagationLevel.Service)
            {
                propagatorSringBuilder.Append("*/");
            }
            else
            {
                // TODO: log warning?
            }
```

Reply:
Unresolving because this is not fixed yet :)

Although you are now validating the setting, the code here should still handle all possible cases.